# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../bin86/no-fortify.patch
# Copyright (C) 2009 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

Compiling the bin86 package in current trunk stops with:

*** buffer overflow detected ***: ncc terminated

when ncc is called for the first time. This is because -D_FORTIFY_SOURCE=2 is 
added to the default CFLAGS in an increasing number of gcc based 
distributions. This enables extra compile-time and run-time checking above and 
beyond what is normal. Obviously, there is a security problem in ncc ;-) But, 
who cares ...

--- dev86-0.16.17/makefile.in.orig	2009-03-05 21:49:12.000000000 +0100
+++ dev86-0.16.17/makefile.in	2009-03-05 21:41:48.000000000 +0100
@@ -42,7 +42,7 @@
 WALL  =-Wall -Wstrict-prototypes
 
 CC	=%CC%
-CFLAGS	=$(GCCFLAG) $(WALL) -O2 -g
+CFLAGS	=$(GCCFLAG) $(WALL) -O2 -g -D_FORTIFY_SOURCE=0
 #endif
 
 #ifndef GNUMAKE
